pgjdbc/src/main/java/org/postgresql/core/Parser.java

```diff
@@ -1163,110 +1163,32 @@ private static String escapeFunction(String functionName, String args, boolean s
     }
   }
 
+  private final static char[] QUOTE_OR_ALPHABETIC_MARKER = new char[]{'\"', '0'};
+  private final static char[] SINGLE_QUOTE = new char[]{'\''};
+
   // Static variables for parsing SQL when replaceProcessing is true.
   private enum SqlParseState {
-    IN_SQLCODE {
-      char[] keyword = new char[0];
-
-      @Override
-      char[] escapeKeyword() {
-        return keyword;
-      }
-    },
-    ESC_DATE {
-      char[] keyword = new char[]{'d'};
-      String replacementKeyword = "DATE ";
-
-      @Override
-      char[] escapeKeyword() {
-        return keyword;
-      }
-
-      @Override
-      String replacementKeyword() {
-        return replacementKeyword;
-      }
-    },
-    ESC_TIME {
-      char[] keyword = new char[]{'t'};
-      String replacementKeyword = "TIME ";
-
-      @Override
-      char[] escapeKeyword() {
-        return keyword;
-      }
-
-      @Override
-      String replacementKeyword() {
-        return replacementKeyword;
-      }
-    },
-    ESC_TIMESTAMP {
-      char[] keyword = new char[]{'t','s'};
-      String replacementKeyword = "TIMESTAMP ";
-
-      @Override
-      char[] escapeKeyword() {
-        return keyword;
-      }
-
-      @Override
-      String replacementKeyword() {
-        return replacementKeyword;
-      }
-    },
-    ESC_FUNCTION {
-      char[] keyword = new char[]{'f','n'};
-
-      @Override
-      char[] escapeKeyword() {
-        return keyword;
-      }
-
-      @Override
-      char[] allowedValues() {
-        return QUOTE_OR_ALPHABETIC_MARKER;
-      }
-    },
-    ESC_OUTERJOIN {
-      char[] keyword = new char[]{'o','j'};
-
-      @Override
-      char[] escapeKeyword() {
-        return keyword;
-      }
-
-      @Override
-      char[] allowedValues() {
-        return QUOTE_OR_ALPHABETIC_MARKER;
-      }
-    },
-    ESC_ESCAPECHAR {
-      char[] keyword = new char[]{'e','s','c','a','p','e'};
-      String replacementKeyword = "ESCAPE ";
-
-      @Override
-      char[] escapeKeyword() {
-        return keyword;
-      }
-
-      @Override
-      String replacementKeyword() {
-        return replacementKeyword;
-      }
-    };
+    IN_SQLCODE,
+    ESC_DATE(new char[]{'d'}, SINGLE_QUOTE, "DATE "),
+    ESC_TIME(new char[]{'t'}, SINGLE_QUOTE, "TIME "),
 
-    private final static char[] QUOTE_OR_ALPHABETIC_MARKER = new char[]{'\"', '0'};
-    private final static char[] SINGLE_QUOTE = new char[]{'\''};
+    ESC_TIMESTAMP(new char[]{'t','s'}, SINGLE_QUOTE, "TIMESTAMP "),
+    ESC_FUNCTION(new char[]{'f','n'}, QUOTE_OR_ALPHABETIC_MARKER, null),
+    ESC_OUTERJOIN(new char[]{'o','j'}, QUOTE_OR_ALPHABETIC_MARKER, null),
+    ESC_ESCAPECHAR(new char[]{'e','s','c','a','p','e'}, SINGLE_QUOTE, "ESCAPE ");
```

Comment:
hm reason why I changed that, was to avoid unnecessary String allocation (even that it's static). but if it's ok for you, i'll change it back to `.toCharArray()`